File: Programmierung II/Uebungsblatt2/PhoneNumber.java
package Uebungsblatt2;

import java.util.Scanner;

public class PhoneNumber {
    private String laenderVorwahl, netzwerkVorwahl, nummer;

    PhoneNumber() {
        laenderVorwahl = getUserInput(2, 3, "Ländervorwahl");
        netzwerkVorwahl = getUserInput(3, 4, "Netzwerkvorwahl");
        nummer = getUserInput(8, 9, "Nummer");
    }

    public String getUserInput(int start, int end, String title) {
        String input;
        do {
            System.out.print(
                    "Bitte geben Sie eine " + title + " mit einer Länge zwischen " + start + " und " + end + " ein: ");
            Scanner scanner = new Scanner(System.in);
            input = scanner.nextLine();
        } while (input.length() < start || input.length() > end);
        return input;
    }

    public String getLeanderVorwahl() {
        return laenderVorwahl;
    }

    public String getNetzwerkVorwahl() {
        return netzwerkVorwahl;
    }

    public boolean equals(PhoneNumber phoneNumber) {
        return laenderVorwahl.equals(phoneNumber.getLeanderVorwahl())
                && netzwerkVorwahl.equals(phoneNumber.getNetzwerkVorwahl());
    }

    public void ausgabe(int mode) {
        switch (mode) {
            case 0:
                System.out.println("+" + laenderVorwahl + " " + netzwerkVorwahl + " " + nummer);
                break;
            case 1:
                System.out.println(0 + netzwerkVorwahl + " " + nummer);
                break;
            default:
                System.out.println("Falscher Modus");
                break;
        }
    }

    public static void main(String[] args) {
        PhoneNumber phoneNumber1 = new PhoneNumber();
        PhoneNumber phoneNumber2 = new PhoneNumber();
        System.out.println(phoneNumber1.equals(phoneNumber2));
        phoneNumber1.ausgabe(0);
        phoneNumber1.ausgabe(1);
    }

}

File: Programmierung II/Uebungsblatt2 und teilweise 3/PhoneNumber.cpp
#include <iostream>
#include <cstring>
using namespace std;

class PhoneNumber
{
private:
    char laenderVorwahl[3];
    char netzwerkVorwahl[4];
    char nummer[9];

public:
    PhoneNumber();
    const char *getLeanderVorwahl() const;
    const char *getNetzwerkVorwahl() const;
    bool equals(const PhoneNumber &) const;
    void ausgabe(bool) const;
    char *getUserInput(int, int, char *);
};

PhoneNumber::PhoneNumber()
{
    strcpy(laenderVorwahl, getUserInput(2, 3, "Ländervorwahl"));
    strcpy(netzwerkVorwahl, getUserInput(3, 4, "Netzwerkvorwahl"));
    strcpy(nummer, getUserInput(8, 9, "Nummer"));
}

const char *PhoneNumber::getLeanderVorwahl() const
{
    return laenderVorwahl;
}

const char *PhoneNumber::getNetzwerkVorwahl() const
{
    return netzwerkVorwahl;
}

bool PhoneNumber::equals(const PhoneNumber &other) const
{
    return strcmp(laenderVorwahl, other.laenderVorwahl) == 0 && strcmp(netzwerkVorwahl, other.netzwerkVorwahl) == 0 && strcmp(nummer, other.nummer) == 0;
}

void PhoneNumber::ausgabe(bool mode = 0) const
{
    if (mode == 1)
    {
        cout << "+" << laenderVorwahl << " " << netzwerkVorwahl << " " << nummer << endl;
    }
    else if (mode == 0)
    {
        cout << 0 << netzwerkVorwahl << " " << nummer << endl;
    }
    else
    {
        cout << "Falscher Modus" << endl;
    }
}

char *PhoneNumber::getUserInput(int start, int end, char *title)
{
    char *input = new char[end + 1];
    do
    {
        cout << "Bitte geben Sie eine " << title << " mit einer Länge zwischen " << start << " und " << end << " ein: ";
        cin >> input;
    } while (strlen(input) < start || strlen(input) > end);
    return input;
}

int main()
{
    PhoneNumber num1;
    PhoneNumber num2;

    num1.ausgabe();
    num2.ausgabe();

    return 0;
}
